node/src/node.rs

```diff
@@ -54,7 +53,7 @@ where
     SyncerSrv: SyncerService<P2pSrv>,
 {
     pub async fn new(config: NodeConfig) -> Result<Self, NodeError<P2pSrv, SyncerSrv>> {
-        let store = Arc::new(RwLock::new(Store::new()));
+        let store: Arc<BoxedStore> = Arc::new(Box::new(InMemoryStore::new()));
```

Comment:
Store should be configurable, so add the following in the `NodeConfig`, without the `Arc`. Wrap it in `Arc` in this method.

Also since this type went all the way up to the `Node` constructor, I started thinking that maybe it shouldn't be boxed and be generic. Box is nice if you want to hide the abstraction from the upper layer, but we are going to expose it because we want it configurable.

Reply:
`Store` implementation is now passed via `NodeConfig`